scheds/rust/scx_rusty/src/bpf/main.bpf.c

```diff
@@ -860,10 +913,19 @@ void BPF_STRUCT_OPS(rusty_stopping, struct task_struct *p, bool runnable)
 {
 	struct task_ctx *taskc;
 
-	if (fifo_sched)
+	if (!(taskc = lookup_task_ctx(p)))
 		return;
+		
+	s32 cpu = scx_bpf_task_cpu(p);
+	struct cpu_task_data *ctd;
 
-	if (!(taskc = lookup_task_ctx(p)))
+	ctd = MEMBER_VPTR(cpu_task_data, [cpu]);
+	if (ctd) {
+		ctd->has_running_task = false;
```

Comment:
This lookup is a bit racy in that another task with low vruntime could be scheduled on the core shortly after, but I'm not sure if there's any way around that. In general it feels like it's mimicking the behavior of the idle cpumask already. We haven't been able to find an idle core in `rusty_select_cpu()` by the time we get to the preemption logic, so maybe we should just forget this and assume that every core has a task? We'd probably need to implement the peek operation as Tejun suggested.

Reply:
If we base the preemption decision on the starting vtime - ie. a task can preempt a cpu iff the preempting task's vtime is prior to the preemptee's vtime when it started executing on the cpu, we'd be a bit more conservative with preemption but always correct and wouldn't need peeking, right?

Reply:
Yes indeed, that sounds better

Reply:
> We haven't been able to find an idle core in rusty_select_cpu() by the time we get to the preemption logic, so maybe we should just forget this and assume that every core has a task?

Updating the vtime for the CPU (in the `running` callback) will be racy as well right? Maybe you're saying that if we are even checking the vtime of the task that just went off the CPU, it's still probably better than not checking as the now outdated vtime might be a decent indication for the vtimes of the other tasks in the dsq.